tests/anomaly_detection/test_half_space_trees.py

```diff
@@ -0,0 +1,33 @@
+from array import array
+
+import numpy as np
+
+from skmultiflow.anomaly_detection import HalfSpaceTrees
+from skmultiflow.data import SEAGenerator
+
+
+def test_half_space_trees():
+    np.random.seed(1)
+    stream = SEAGenerator(classification_function=0, noise_percentage=0.1)
+    stream.prepare_for_use()
+    learner = HalfSpaceTrees(n_features=stream.n_features, n_estimators=25, size_limit=25, anomaly_threshold=0.90,
+                             depth=10, tree_random_state=5)
+
+    cnt = 0
+    max_samples = 5000
+    y_pred = array('i')
+    y_proba = []
+    wait_samples = 500
+
+    while cnt < max_samples:
+        X, y = stream.next_sample()
+        # Scale inputs between 0 and 1
+        X = X / 10
+        if (cnt % wait_samples == 0) and (cnt != 0):
+            y_pred.append(learner.predict(X)[0])
+            y_proba.append(learner.predict_proba(X)[0])
```

Comment:
This line serves for coverage. However, given its relevance, it is important to have an explicit check on the actual probability values.

Something in the lines of:

```python
assert np.allclose(y_proba, expected_proba)
```

You can explicitly write the expected probability values (`expected_proba`) or save them in a `numpy` file to be loaded by the test.

Reply:
> This line serves for coverage. However, given its relevance, it is important to have an explicit check on the actual probability values.
> 
> Something in the lines of:
> 
> ```python
> assert np.allclose(y_proba, expected_proba)
> ```
> 
> You can explicitly write the expected probability values (`expected_proba`) or save them in a `numpy` file to be loaded by the test.

I have included the change, however probabilities are a bit different and test is not passing every time. I will try to find the root of this problem.
For now, I am just sharing the result:

```
expected_proba = array([[7.81250000e-06, 9.99992187e-01],
       [1.61760000e-01, 8.38240000e-01],
       [1.98240000e-01, 8.01760000e-01],
       [2.19360000e-01, 7.80640000e-01],
       [1.72000000e-01, 8.28000000e-01],
       [7.29837500e-02, 9.27016250e-01],
       [2.16320000e-01, 7.83680000e-01],
       [2.13920000e-01, 7.86080000e-01],
       [2.07373125e-01, 7.92626875e-01]])
```
```
y_proba    = [array([7.81250000e-06, 9.99992187e-01]),
 array([0.14832, 0.85168]),
 array([0.18864, 0.81136]),
 array([0.20784, 0.79216]),
 array([0.16384, 0.83616]),
 array([0.07984, 0.92016]),
 array([0.20784, 0.79216]),
 array([0.20464, 0.79536]),
 array([0.21568, 0.78432])]
```

Reply:
> > This line serves for coverage. However, given its relevance, it is important to have an explicit check on the actual probability values.
> > Something in the lines of:
> > ```python
> > assert np.allclose(y_proba, expected_proba)
> > ```
> > 
> > 
> > You can explicitly write the expected probability values (`expected_proba`) or save them in a `numpy` file to be loaded by the test.
> 
> I have included the change, however probabilities are a bit different and test is not passing every time. I will try to find the root of this problem.
> For now, I am just sharing the result:
> 
> ```
> expected_proba = array([[7.81250000e-06, 9.99992187e-01],
>        [1.61760000e-01, 8.38240000e-01],
>        [1.98240000e-01, 8.01760000e-01],
>        [2.19360000e-01, 7.80640000e-01],
>        [1.72000000e-01, 8.28000000e-01],
>        [7.29837500e-02, 9.27016250e-01],
>        [2.16320000e-01, 7.83680000e-01],
>        [2.13920000e-01, 7.86080000e-01],
>        [2.07373125e-01, 7.92626875e-01]])
> ```
> 
> ```
> y_proba    = [array([7.81250000e-06, 9.99992187e-01]),
>  array([0.14832, 0.85168]),
>  array([0.18864, 0.81136]),
>  array([0.20784, 0.79216]),
>  array([0.16384, 0.83616]),
>  array([0.07984, 0.92016]),
>  array([0.20784, 0.79216]),
>  array([0.20464, 0.79536]),
>  array([0.21568, 0.78432])]
> ```

So, I found the root of the problem, which was basically creation of trees before setting the `random_state`. I also added `from skmultiflow.utils import check_random_state` instead of setting random seed manualy in the code.

Reply:
Awesome! Just in case, you can look here `src/skmultiflow/meta/classifier_chains.py` for an example on the standard way we use `random_state`.

Notice in the `__init__` method that there are two parameters:

```python
        self.random_state = random_state    # This parameters has the same name as the parameter in the method's signature
        self._random_state = None           # This is the actual random_state object used internally
```

`self.random_state` is used by the `get_info` method and `reset` methods and it only keeps the **value** that was passed by the user. `self._random_state` is the actual `random` **object** used internally. It is important to make sure that you **only** use `self._random_state` internally.